app/scripts/filters/util.js

```diff
@@ -528,6 +528,36 @@ angular.module('openshiftConsole')
       }
       var uri = new URI(url);
       var protocol = uri.protocol();
-      return uri.is('absolute') && (protocol === 'http' || protocol === 'https');      
+      return uri.is('absolute') && (protocol === 'http' || protocol === 'https');
+    };
+  })
+  // currently used to set an alt text value for an env var that has a
+  // valueFrom attribute instead of a value.
+  .filter('altTextForValueFrom', function() {
+    return function(envVar) {
+      if(envVar.value) {
+        return;
+      }
+      if(!envVar.valueFrom) {
+        return;
+      }
+      envVar.valueIcon = 'pficon pficon-help';
+      envVar.valueIconTooltip = 'This is a referenced value that will be generated when a container is created.  On running pods you can check the resolved values by going to the Terminal tab and echoing the environment variable.';
+      _.each(envVar.valueFrom, function(from, refType) {
+        switch(refType) {
+          case 'configMapKeyRef':
+            envVar.valueAlt = 'Set to the key ' + from.key + ' in config map ' + from.name;
```

Comment:
It is, I believe we have used most.  Conceptually I've been thinking of this control a bit like `data` -> `metadata for UI` -> `data`.  Ie, convert the data to a list of metadata for controlling the UI, interact with it, then convert it back to data when done.  

I'm good with a function that would copy relevant keys rather than delete irrelevant keys.  I def prefer the "purity" of input/output over the direct manipulation.